File: FileCompressionPython/file_handler.py
import os

class FileHandler:
    
    @staticmethod
    def read_file(filename):
        try:
            with open(filename, 'r') as file:
                return file.read()
        except FileNotFoundError:
            raise FileNotFoundError(f"File '{filename}' not found")
        except IOError as e:
            raise IOError(f"Error reading file '{filename}': {str(e)}")
    
    @staticmethod
    def write_file(filename, content):
        try:
            with open(filename, 'w') as file:
                file.write(content)
        except IOError as e:
            raise IOError(f"Error writing to file '{filename}': {str(e)}")
    
    @staticmethod
    def file_exists(filename):
        return os.path.exists(filename)

File: FileCompressionJava/FileHandler.java
import java.io.*;
import java.nio.file.Files;
import java.nio.file.Paths;

public class FileHandler {
    
    public static String readFile(String filename) throws IOException {
        return new String(Files.readAllBytes(Paths.get(filename)));
    }
    
    public static void writeFile(String filename, String content) throws IOException {
        try (BufferedWriter writer = new BufferedWriter(new FileWriter(filename))) {
            writer.write(content);
        }
    }
    
    public static boolean fileExists(String filename) {
        return Files.exists(Paths.get(filename));
    }
}
